### PR TITLE
have create entity work again

### LINKED ISSUE #1290: Remove schema (Python) in DMT

### LINKED ISSUE #1291: Remove code not used in DMT (probably some things in the API and stuff related to tree node)
domain classes:

1) use of tree_node -> remove

- test_tree_err
- domain.py -> code in this file is not used anywhere else, remove	
- document_service ->  generate_tree() can be removed
- genereate_code_with_plugin_use_case -> can be removed

2) schema.py  -> delete
schema is used by:

- generate_system_blueprints.py -> code not used anywhere, delete
- used in code_genrators... 
- bdd test

3)  dto -  remove

4) home
	DMT/code_generators - docuble check before remove.
	
5) repository 
dmss/TemplateRepositoryForDMSS.py - remove

6) restful
request_object.py , only usedf in code generator.

7) utils
* data_structure folder: 
- dictionary_test.py

only pretty_eq, print_pygments and find function are used by other code...
maybe some of the code in data_structure belongs to utils instead.

dictionary.py can be removed, along with dictionary_test.py

investigate some more...

*  create_entity_utils: 
- CreateEntityException and InvalidDefaultValue delete
- generate_system_blueprints.py is unused

* group_by.py not used - remove

### LINKED ISSUE #1322: Create entity does not create nested objects for MorringSRS?

### PR DESCRIPTION
**Pull request by soofstad**
## What does this pull request change?
- Rework and fix "CreateEntity"
- Remove DTO and clean out other dead code

## Why is this pull request needed?
- CreateEntity did not properly create entities

## Issues related to this change
closes #1322 
closes #1290 
closes #1291

### CODE REVIEW

**Review by KristianKjerstad — CHANGES_REQUESTED** [review]
nice!
Just a minor thing about the integer attribute type

File: api/src/domain_classes/blueprint.py [hunk 1]
```diff
@@ -42,87 +41,20 @@ def get_ui_recipe(recipes: List[Dict]):
...
-            "attributes": [attribute.to_dict() for attribute in self.attributes],
-            "storageRecipes": [recipe.to_dict() for recipe in self.storage_recipes],
-            "uiRecipes": [recipe.to_dict() for recipe in self.ui_recipes],
-        }
-
-    def __eq__(self, other):
-        return self.to_dict() == other.to_dict()
+        self.ui_recipes: List[Recipe] = get_ui_recipe(entity.get("uiRecipes", []))
 
     def get_none_primitive_types(self) -> List[BlueprintAttribute]:
```
Comment (netr0m) [inline api/src/domain_classes/blueprint.py:58]:
```suggestion
    def get_non_primitive_types(self) -> List[BlueprintAttribute]:
```
None-primitive => non-primitive?

File: api/src/services/application_service.py [hunk 2]
```diff
@@ -37,31 +36,29 @@ def get_related_blueprints(self, blueprint: str) -> dict:
...
 
         def get_blueprints_recursive(type: Blueprint):
             for attr in type.get_none_primitive_types():
                 bp = self.document_service.get_blueprint(attr.attribute_type)
-                related_blueprints[attr.attribute_type] = bp.to_dict()
+                related_blueprints[attr.attribute_type] = bp.entity
                 if attr.attribute_type not in related_blueprints.keys():
                     get_blueprints_recursive(bp)
 
         for attr in first.get_none_primitive_types():
```
Comment (netr0m) [inline api/src/services/application_service.py:48]:
```suggestion
        for attr in first.get_non_primitive_types():
```

File: api/src/utils/create_application_utils.py [hunk 3]
```diff
@@ -88,34 +57,33 @@ def zip_all(zip_file: ZipFile, path: str, real_name="", write_folder: bool = Tru
...
+    write_to = f"{path}/{document['name']}.json"
     logger.info(f"Writing: {document['type']} to {write_to}")
 
     if document["type"] != SIMOS.PACKAGE.value:
         ob.writestr(write_to, binary_data)
 
-    blueprint = document_service.get_blueprint(document.type)
+    blueprint = document_service.get_blueprint(document["type"])
     document_references = []
     for attribute in blueprint.get_none_primitive_types():
```
Comment (netr0m) [inline api/src/utils/create_application_utils.py:72]:
```suggestion
    for attribute in blueprint.get_non_primitive_types():
```

**Review by netr0m — CHANGES_REQUESTED** [review]
Looks good! One remark on naming (None-primitive => non-primitive), plus what Kristian mentioned.

File: api/src/tests/unit/instantiate_entity.py [hunk 4]
```diff
@@ -1,45 +1,173 @@
...
+test_blueprints = {
+    "basic_blueprint": {
+        "type": "system/SIMOS/Blueprint",
+        "name": "box",
+        "description": "First blueprint",
+        "attributes": [
+            {"attributeType": "string", "type": "system/SIMOS/BlueprintAttribute", "name": "name"},
+            {"attributeType": "string", "type": "system/SIMOS/BlueprintAttribute", "name": "type"},
+            {"attributeType": "string", "type": "system/SIMOS/BlueprintAttribute", "name": "description"},
+            {"attributeType": "integer", "type": "system/SIMOS/BlueprintAttribute", "name": "length"},
```
[the remark this answers is not in the document]
Comment (soofstad) [reply api/src/tests/unit/instantiate_entity.py:17]: Added "integer" back in. Not sure why I removed it...

File: api/src/utils/create_entity_utils.py [hunk 5]
```diff
@@ -1,119 +1,61 @@
...
-                    if attr.attribute_type == BuiltinDataTypes.OBJECT.value
-                    else self.blueprint_provider(attr.attribute_type)
-                )
-                if not isinstance(blueprint, Blueprint):
-                    blueprint = Blueprint(DTO(blueprint))
-                if attr.is_array():
-                    entity[attr.name] = attr.dimensions.create_default_array(self.blueprint_provider, CreateEntity)
-        return entity
+        case "integer":
+            return 0
```
Comment (netr0m) [inline api/src/utils/create_entity_utils.py:34]: Keeping integer after all?

**Review by netr0m — APPROVED** [review]

**Review by KristianKjerstad — APPROVED** [review]

**Pull request merged**